### ISSUE TITLE
Calling python script with mpirun now does not work 

### ISSUE THREAD

**Issue by alexmascension** [open-mpi/ompi#6773]
Hi,

I am running OpenMPI 4.0.1 on a Kubuntu 19.04 computer.

I have installed OpenMPI in my computer via conda `conda install openmpi`. In order to run mpirun with python I am using mpi4py==3.0.2.

Lastly, I was using OpenMPI v2.1.2 and I usually call parallelized python scripts within a non-parallelized python script. However, this does not happen anymore with OpenMPI v3 and v4.

In order to visualize the problem, I have created two codes: `parallel.py`, which contains the actions to be done by mpirun; and `coder.py`, which calls `parallel.py`. When running `coder.py`, all the key steps are run, except the call to `parallel.py`. When running `parallel.py` alone, the code runs normally.

`parallel.py`
```python
from mpi4py import MPI
comm = MPI.COMM_WORLD
print('RANK %s'%comm.rank)
```

`coder.py`
```python
import os, sys
syspath = os.path.dirname(os.path.realpath(__file__))

print('Running code')
com = 'mpirun -v -np 3 ' + str(sys.executable) + " -m mpi4py " + syspath + "/parallel.py"
os.system(com)
print('Done')
```

After calling `python coder.py`:

EXPECTED (OpenMPI v2.1.2)
```
Running code
RANK 1
RANK 0
RANK 2
Done
```

OBSERVED
```
Running code
Done
```
The same happens if I run `subprocess.Popen`: `stdout=''` and `stderr=''` are returned.

After calling `mpirun -np 3 python parallel.py`:
```
RANK 1
RANK 0
RANK 2
```
I guess that by calling `os.system` after python the process is already "blocked" and `os.system` is unable to call mpirun because of this "blocking". However, this has to be something from the version v3 and v4 because with v2.1.2 I was able to do that.

Thank you for your support

**Comment by ggouaillardet** [comment]
are you sure `mpirun` matches the `libmpi.so` used by `mpi4py` ?

**Comment by alexmascension** [comment]
How could I check that with a python command? I guess still that the matching is correct because running `mpirun -np 3 python parallel.py` from console does indeed run the program correctly. Is it possible that calling `os.system()` or `subprocess.Popen()` from python alters the environment by which mpirun and mpi4py is called?

**Comment by ggouaillardet** [comment]
You can `type mpirun` and then `mpirun hostname` instead of a mpi4py app. If it works, then try with a `C` MPI app.

**Comment by jsquyres** [comment]
I have seen that mpi4py needs to be compiled with the exact version of Open MPI that you are using.  E.g., if you have upgraded Open MPI, you need to re-compile mpi4py (depending on how you installed mpi4py, you may need to uninstall it and re-install it to force it to recompile against your new version of Open MPI).

**Comment by alexmascension** [comment]
Hi,

I have uninstalled everything and installed `mpi4py` 3.0.1 via conda, which installs OpenMPI 3.1.4. The error continued; so I decided to create a new conda environment, and installing mpi4py automatically installed mpi 1.0 and mpich 3.2.1. Now the problem has been solved.

I uploaded a docker image to replicate the success if you want to docker: `docker pull alexmascension/mpirun-test`

Thanks for the help!
